src/lib/layouts/DocsTutorial.svelte

```diff
@@ -31,13 +32,43 @@
 
     let slotContent: HTMLElement | null = null;
 
+    /**
+     * Due to underlying logic with anchor links & the auto-scroll via hash values in the URL,
+     * we have an issue where if the first item is not scrolled enough it isn't marked as `selected`.
+     *
+     * We do below workaround for the time being without breaking things to scroll to the first item.
+     */
+    async function preSelectItemOnInit() {
+        await tick();
+
+        if (!$page.url.hash) return;
+        const tocItem = toc.slice(1);
+
+        // no sub-items, return.
+        if (!tocItem.length) return;
+
+        const pageHash = $page.url.hash.replace('#', '');
+        const tocItemHref = tocItem[0].href.replace('#', '');
+
+        if (pageHash !== tocItemHref) return;
+
+        const element = document.getElementById(pageHash);
+        if (element) {
+            const offset = 50;
+            const rect = element.getBoundingClientRect();
+            window.scroll({ top: window.scrollY + rect.top - offset });
```

Comment:
Can you also use https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView for this?

Reply:
I did that earlier with multiple options against `scrollInToView`, that doesn't help. Its the same behaviour as the anchor link visits on hash urls. And that doesn't scroll enough, we still need some offset to make the header be covered enough so its marked as `selected`.